rubicon/objc/objc.py

```diff
@@ -1644,3 +1644,23 @@ def finalize(self, cmd) -> None:
             anObject = get_instance_variable(self, 'observed_object', objc_id)
             ObjCInstance._cached_objects.pop(anObject, None)
             send_super(self, 'finalize')
+
+
+class ObjCBlockStruct(Structure):
+    _fields_ = [
+        ('isa', c_void_p),
+        ('flags', c_int),
+        ('reserved', c_int),
+        ('invoke', CFUNCTYPE(c_void_p, c_void_p)),
+    ]
+
+
+class ObjCBlock:
```

Comment:
Update: After discussing this on Gitter, we found out that this isn't as easy as I thought.

Because of `ObjCInstance`'s caching mechanism, it's not possible to have two `ObjCInstance`s for the same object, even if they have different classes. This means that in this case the `ObjCBlock` constructor would return a normal `ObjCInstance` if one already existed for the object (which is very common).

It's not possible to handle blocks inside the `ObjCInstance` constructor, like we do with `ObjCClass`, `ObjCListInstance`, etc., because there is no way to see at runtime whether an object is a block or not.